code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -12,6 +12,7 @@
 	var/silent = 0 // No message on putting items in
 	var/list/can_hold = new/list() //List of objects which this item can store (if set, it can't store anything else)
 	var/list/cant_hold = new/list() //List of objects which this item can't store (in effect only if can_hold isn't set)
+	var/empty = FALSE // Will this spawn as an empty box
```

Comment:
I dont like this being made just for donut boxes. Either refactor all empty storage boxes to use this var, or remove it.

Reply:
As far as I can tell the only way to avoid using an `empty` variable for donut boxes would be to make `/obj/item/storage/fancy/donut_box` the empty one, and add a `filled` subtype.
That of course is out of the question since it would need map edits, so I guess I'll get started on a storage refactor.

Reply:
Don't use a variable for this. Just make a separate method called during `New()` that fills a storage item. 

Way easier to override and no hidden quirks.